controller/iteration.go

```diff
@@ -232,6 +232,92 @@ func (c *IterationController) Update(ctx *app.UpdateIterationContext) error {
 	})
 }
 
+// Delete runs the delete action.
+func (c *IterationController) Delete(ctx *app.DeleteIterationContext) error {
+	currentUser, err := login.ContextIdentity(ctx)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, goa.ErrUnauthorized(err.Error()))
+	}
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		itr, err := appl.Iterations().Load(ctx.Context, ctx.IterationID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		s, err := appl.Spaces().Load(ctx, itr.SpaceID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
+		}
+		if !uuid.Equal(*currentUser, s.OwnerId) {
+			log.Warn(ctx, map[string]interface{}{
+				"space_id":     s.ID,
+				"space_owner":  s.OwnerId,
+				"current_user": *currentUser,
+			}, "user is not the space owner")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not the space owner"))
+		}
+		if itr.IsRoot(s.ID) {
+			log.Warn(ctx, map[string]interface{}{
+				"space_id":     s.ID,
+				"iteration_id": itr.ID,
+			}, "can not delete root iteration")
+			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("can not delete root iteration"))
+		}
+		subtree, err := appl.Iterations().LoadChildren(ctx, ctx.IterationID)
+		if err != nil {
+			return jsonapi.JSONErrorResponse(ctx, err)
+		}
+		// Fetch parent iteration to which work items will get attached
+		parentID := itr.Parent()
+		if parentID == uuid.Nil {
```

Comment:
Is this a real scenario?

Reply:
no, not actually, but the function returns that uuid.NIL in some cases hence I need to check it. Part of error handling